thunder/core/jit_ext.py

```diff
@@ -1054,6 +1054,8 @@ def is_from_torch(fn):
 
         has_tensor_arg = False
         for a in args:
+            if is_opaque(fn) and isinstance(a.value, thunder.dtypes.dtype):
+                a.value = thunder.dtypes.to_torch_dtype(a.value)
```

Comment:
Unfortunately, I think this was a red herring, sorry about that! We need to have something after unwrapping, which is not here.
With that, I think it might be better to have a `finfo` function in thunder/torch/__init__.py that just does the thunder dtype -> torch.dtype conversion, calls `torch.finfo` and returns the result (not sure if we need to handle the result type but I think we might try as is).